hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/schemas/GenesisSchema.java

```diff
@@ -273,11 +276,35 @@ private void createGenesisExchangeRate(
     // Creates and loads the network properties into state
 
     private void createGenesisNetworkProperties(
-            @NonNull final BootstrapConfig v,
+            @NonNull final BootstrapConfig bootstrapConfig,
             @NonNull final FilesConfig filesConfig,
-            @NonNull final WritableKVState<FileID, File> files) {
+            @NonNull final WritableKVState<FileID, File> files,
+            @NonNull final Configuration configuration) {
         logger.debug("Creating genesis network properties file");
-        // TBD Implement this method
+
+        // Get the set of network properties from configuration, and generate the file content to store in state.
+        List<Setting> settings = new ArrayList<>();
+        Utils.networkProperties(configuration)
+                .forEach((propertyName, propertyValue) -> settings.add(Setting.newBuilder()
+                        .name(propertyName)
+                        .value(propertyValue.toString())
+                        .build()));
+
+        final var servicesConfigList =
+                ServicesConfigurationList.newBuilder().nameValue(settings).build();
+        final var fileNum = filesConfig.networkProperties();
+        // default to shard=0, realm=0. Should get from config
+        final var fileId = FileID.newBuilder().fileNum(fileNum).build();
+        final var masterKey =
+                Key.newBuilder().ed25519(bootstrapConfig.genesisPublicKey()).build();
+        files.put(
+                fileId,
+                File.newBuilder()
+                        .contents(ServicesConfigurationList.PROTOBUF.toBytes(servicesConfigList))
+                        .fileId(fileId)
+                        .keys(KeyList.newBuilder().keys(masterKey))
+                        .expirationTime(bootstrapConfig.systemEntityExpiry())
+                        .build());
```

Comment:
No unit test for this generated schema (or anything else in this file I think).  Is there _any_ kind of test (e.g., itest/e2e) that things are working correctly?  Or is it that the whole system drops dead on starting if these files are wrong? Or?

Reply:
True, it needs tests. I'm going to let the owner implement the unit tests. This is getting tested in the e2e/bdd tests.